### PR TITLE
OOP Library Setup Associations

### PR DESCRIPTION
**Pull request by Munyabelden**
### About This Pull Request

- [ ] Created Classroom class with `@label`  variables
- [ ] Create Book class.
- [ ] Created Rental class
- [ ] Implemented the one-to-many association between Classroom and students
- [ ] Implemented many-to-many relationship between Person and Book classes 

**Thank you for the review🤍**

### CODE REVIEW

File: classroom.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class Classroom
+  def initialize(label)
+    @label = label
+    @students = []
+  end
+
+  attr_accessor :label, :students
```
Comment (mikethreels) [inline classroom.rb]: - [ ] There is no need to change the `student` instance variable from outside the Classroom. therefor please use the attribute reader for this variable.

File: student.rb [hunk 2]
```diff
@@ -5,7 +5,13 @@ class Student < Person
 
   def initialize(age, name = 'unknown', parent_permission: true, classroom: nil)
     super(age, name, parent_permission: parent_permission)
+    self.classroom = classroom
+  end
+
+  def classroom=(classroom)
+    @classroom&.students&.delete(self)
     @classroom = classroom
+    classroom.students << self if classroom
```
Comment (mikethreels) [inline student.rb]: - [ ] Good job adding the student to classroom.students however it should be added unless it the student is already in the classroom

File: person.rb [hunk 3]
```diff
@@ -1,14 +1,31 @@
...
     @parent_permission = parent_permission
+    @rentals = []
+  end
+
+  def correct_name
+    @name
   end
 
-  attr_reader :id
+  attr_reader :id, :rentals
```
Comment (mikethreels) [inline person.rb]: - [ ] in the person class you should also be able to change the rentals from outside the class, please add it therefor to the attr_accessor.

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Belden, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- good job setting up the associations.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by OLIPLICHE — APPROVED** [review]
Hi @Munyabelden, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Highlights
- Applied the changes as requested by the previous review👏
- Created the remaining classes and associations between them👏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**